File: apps/android/app/src/main/java/com/pcmobilelink/nearshare/pairing/PairingPayload.kt
package com.pcmobilelink.nearshare.pairing

data class PairingPayload(
    val version: Int,
    val offerId: String,
    val pcName: String,
    val endpoints: List<PairingEndpointCandidate>,
    val pairingToken: String,
    val shortCode: String? = null,
    val tlsCertificateSha256: String,
    val expiresAtUnixTimeSeconds: Long,
    val transport: String,
)

File: apps/windows/src/PCMobileLink.Windows.Core/PairingPayload.cs
namespace PCMobileLink.Windows.Core;

public sealed record PairingPayload
{
    public int Version { get; init; } = 1;

    public required Guid OfferId { get; init; }

    public required string PcName { get; init; }

    public required IReadOnlyList<PairingEndpointCandidate> Endpoints { get; init; }

    public required string PairingToken { get; init; }

    public string? ShortCode { get; init; }

    public required string TlsCertificateSha256 { get; init; }

    public required long ExpiresAtUnixTimeSeconds { get; init; }

    public string Transport { get; init; } = "https";
}
